lib/install/deps.js

```diff
@@ -451,10 +451,13 @@ function failedDependency (tree, name, pkg) {
 
   if (!tree.requiredBy) return false
 
+  doneList = doneList || {}
+  doneList[tree] = true
+
   let anyFailed = false
   for (var ii = 0; ii < tree.requiredBy.length; ++ii) {
     var requireParent = tree.requiredBy[ii]
-    if (failedDependency(requireParent, moduleName(tree), tree)) {
+    if (!doneList[requireParent] && failedDependency(requireParent, moduleName(tree), tree, done)) {
```

Comment:
this should be

```if (!doneList[requireParent] && failedDependency(requireParent, moduleName(tree), tree, doneList)) {```

Reply:
Thanks. I fixed this by using `done` everywhere, to match the coding style of `loadRequestedDeps()` or `getAllMetadata` in the same file.

Reply:
done is actually a keyword used for callback in most js program so it was better with the older variable name.

I'm not a maintainer of this project, and it doesn't seems like one of them is looking at this PR anyway